### ISSUE TITLE
Innapropriate 503 message error when search someone in unknowned instance

### ISSUE THREAD

**Issue by GuyM-31** [mastodon/mastodon#7832]
Hi,

Please apologise my poor English.

If you search people in an unowned instance (for example a@aa) an error message (white on black) appear rapidly "503Remote data could not be fetched" or "503Remote SSL certificate could not be verified"
This message lets think that your instance have problems. (503 http error means 503 Service Unavailable, which is false in this case).

The return Have To be at least "0 result" (as search without results) or better "0 result @aa instance error : wrong name or unreachable"

I hope that I have been understandable.

I hope this will help to make Mastodon greater.

GM

**Comment by alexg-k** [comment]
So what is the reason for this error then? In my case the user searched does exist.

**Comment by pfefferle** [comment]
I have two cases (using the [WordPress ActivityPub plugin](https://github.com/pfefferle/wordpress-activitypub)) with the same issue:

* https://github.com/pfefferle/wordpress-activitypub/issues/11
* https://chaos.social/@crossgolf_rebel/101411114032959361

SSL is working in both cases, WebFinger identifiers seems to be fine and profile URLs also.

**Comment by sebilasse** [comment]
Same here.

**Comment by SuperSandro2000** [comment]
related to #6449 ?

**Comment by GuyM-31** [comment]
No

**Comment by SuperSandro2000** [comment]
I had that issue to and I removed all ipv6 dns entries from my domain and turned on tls 1.2. Before that I only allowed tls 1.3.

**Comment by cgarwood82** [comment]
I think this may be related to DNS in some way. My configuration is I've got a peertube(vid.garwood.io) instance that shares a reverse proxy with my mastodon instance(social.garwood.io). Both work perfectly independent of each other. But I've found that trying to follow my peertube user from my mastodon server gives me all sorts of issues. 

I've got two different configurations that generate different results, both of which are induced by local dns changes. 

If I create an internal record for the public DNS endpoint to point to a reverse proxy address, I get the following error from the mastodon UI:
`422 Mastodon::HostValidationError on https://vid.garwood.io/accounts/clifford`

If I delete the above record created on my local resolver and instead add the same exact record directly to the reverse proxy which is shared between both peertube and mastodon, I get the 503 message from the UI:
`503 Remote SSL certificate could not be verified`

For what it is worth, I was able to follow the pertube account from another mastodon account outside of mastodon server that shares the same web proxy. All certs are valid on the reverse proxy. The only thing I can imagine is that when searching for an account, it's mangling the name somehow. 

I'm not terribly familiar with the code base here, but if someone can point me to relevant code concerning searches, I'll gladly take a look and see if I can't work something out.
